src/FastJetTopTagger.cpp

```diff
@@ -286,6 +362,22 @@ void FastJetTopTagger::processEvent(LCEvent * evt){
   
 } //end processEvent
 
+double FastJetTopTagger::getECF(PseudoJet& jet, int whichECF, double beta, std::string energyCorrelator){
+  /// ********** Energy Correlation Function *** /////////
+
+  // options for EnergyCorrelator:
+  //   pt_R (transverse momenta and boost-invariant angles)
+  //   E_theta (energy and angle as dot product of vectors)
+  //   E_inv (energy and angle as (2p_i \cdot p_j/E_i E_j)
+
+  if(!jet.has_constituents())
+    return -1.0;
+    
+  EnergyCorrelator ECF(whichECF, beta, _energyCorrMap[energyCorrelator]);
```

Comment:
The objects can be instantiated once, and then used, instead of creating one every time the function is called

Reply:
done